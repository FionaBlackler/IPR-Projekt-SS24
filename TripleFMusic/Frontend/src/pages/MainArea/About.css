.about-container {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    width: 100%; /* Seitenübergreifend */
  }

  .window {
    width: 100%; /* Fensterbreite */
  }

  .window-header {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .content {
    text-align: center;
  }

  .cutout {
    margin: 1em 0;
    padding: 1em;
    width: 100%; /* Volle Breite */
    text-align: center; /* Text zentrieren */
  }

  .about-section {
    margin-bottom: 1em;
    width: 100%; /* Volle Breite */
  }

  .section-header {
    text-align: center;
  }

  .team-container {
    display: flex;
    justify-content: space-around;
    align-items: center;
    width: 100%; /* Volle Breite */
  }

  .team-member {
    text-align: center;
  }

  .team-member img {
    width: 150px; /* Rechteckige Bilder */
    height: 200px;
    margin-bottom: 0.5em;
    object-fit: cover; /* Zuschneiden des Bildes, um das Rechteck zu füllen */
  }
