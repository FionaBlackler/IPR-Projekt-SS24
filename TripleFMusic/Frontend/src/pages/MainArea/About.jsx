import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Window, WindowHeader, WindowContent, Panel, Cutout, Button, Tooltip } from 'react95';
import rose from 'react95/dist/themes/rose';
import './About.css';

const About = () => {
  return (
    <ThemeProvider theme={rose}>
      <div className="about-container">
        <Window className="window">
          <WindowHeader className="window-header">
            <span>About TripleF Music</span>
          </WindowHeader>
          <WindowContent>
            <Panel variant="well" className="about-section">
              <h2 className="section-header">About us</h2>
              <Cutout className="cutout">
                <p>Welcome to TripleF Music! 
                  We are a passionate team of three students from HFT Stuttgart, 
                  dedicated to bringing you a unique and engaging music experience. 
                  Our journey began with a shared love for music and technology, 
                  driving us to create a platform where users can explore, discover, 
                  and enjoy music in an oldschool and exciting way. Join us on our journey 
                  to revolutionize the way you experience music.</p>
              </Cutout>
            </Panel>

            <Panel variant="well" className="about-section">
              <h2 className="section-header">Our vision and goal</h2>
              <Cutout className="cutout">
                <p>We aim to bring back the old-school nostalgia, 
                  giving users a delightful time-travel experience to the days when life was 
                  simpler and music had that special charm. Our main goal? Well, it's to pass 
                  our module, to be honest! But hey, a bit of creativity and fun never hurt anyone, 
                  right? We're here to combine our passion for retro vibes with modern technology, 
                  crafting a unique platform that makes discovering and enjoying music feel like a 
                  nostalgic adventure. Join us on this whimsical journey as we blend the best of the 
                  past with the present!</p>
              </Cutout>
            </Panel>

            <Panel variant="well" className="about-section">
              <h2 className="section-header">Our Team</h2>
              <Cutout className="cutout">
                <div className="team-container">
                  <div className="team-member">
                    <img src="/Fatima.png" alt="Fatima" />
                    <p>Fatima</p>
                  </div>
                  <div className="team-member">
                    <img src="/Funda.png" alt="Funda" />
                    <p>Funda</p>
                  </div>
                  <div className="team-member">
                    <img src="/Fiona.png" alt="Fiona" />
                    <p>Fiona</p>
                  </div>
                </div>
              </Cutout>
            </Panel>

            <Panel variant="well" className="about-section">
              <h2 className="section-header">Our motivation</h2>
              <Cutout className="cutout">
          <Tooltip text='Lets tackle this module like its a piÃ±ata filled with passing grades â€“ swing hard and aim for success! ðŸ¤·' enterDelay={100} leaveDelay={200} position='top'>
                 <div className='matze'>
                  <Button>Matze</Button>
                </div>
              </Tooltip>
            </Cutout>
          </Panel>

            <Panel variant="well" className="about-section">
              <h2 className="section-header">Contact</h2>
              <Cutout className="cutout">
                <p>Email: info@triplefmusic.de</p>
                <p>Telefon: +49 171 56920382</p>
              </Cutout>
            </Panel>
          </WindowContent>
        </Window>
      </div>
    </ThemeProvider>
  );
}

export default About;
