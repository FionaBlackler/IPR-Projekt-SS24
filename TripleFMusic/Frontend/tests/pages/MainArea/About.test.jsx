import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import About from "../../../src/pages/MainArea/About";

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

describe('About Component', () => {
  const navigate = jest.fn();

  beforeEach(() => {
    require('react-router-dom').useNavigate.mockReturnValue(navigate);
  });

  const renderComponent = () => {
    return render(
      <MemoryRouter>
        <About />
      </MemoryRouter>
    );
  };

  test('renders About component correctly', () => {
    renderComponent();
    expect(screen.getByText(/About us!/i)).toBeInTheDocument();
    expect(screen.getByText(/Our vision and goal/i)).toBeInTheDocument();
    expect(screen.getByText(/Our team/i)).toBeInTheDocument();
    expect(screen.getByText(/Our motivation/i)).toBeInTheDocument();
    expect(screen.getByText(/Contact/i)).toBeInTheDocument();
  });

  test('navigates when icons are clicked', () => {
    const { getByAltText } = renderComponent();

    fireEvent.click(getByAltText(/About/i));
    expect(navigate).toHaveBeenCalledWith('/welcome/about');

    fireEvent.click(getByAltText(/Home/i));
    expect(navigate).toHaveBeenCalledWith('/welcome/home');

    fireEvent.click(getByAltText(/Music Gallery/i));
    expect(navigate).toHaveBeenCalledWith('/welcome/musicgallery');

    fireEvent.click(getByAltText(/Upload new song/i));
    expect(navigate).toHaveBeenCalledWith('/welcome/addsong');

    fireEvent.click(getByAltText(/Internet Explorer/i));
    expect(navigate).toHaveBeenCalledWith('/welcome/internet');
  });

  test('toggles tooltip text', () => {
    renderComponent();
    const button = screen.getByText(/Matze/i);
    fireEvent.click(button);
    expect(screen.getByText(/Lets tackle this module like its a piñata/i)).toBeInTheDocument();
    fireEvent.click(button);
    expect(screen.queryByText(/Lets tackle this module like its a piñata/i)).not.toBeInTheDocument();
  });

  test('closes About window and navigates to home', () => {
    renderComponent();
    const closeButton = screen.getByRole('button', { name: '' }); // Assuming the close button has no accessible name
    fireEvent.click(closeButton);
    expect(navigate).toHaveBeenCalledWith('/welcome/home');
  });

  test('matches snapshot', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });
});
