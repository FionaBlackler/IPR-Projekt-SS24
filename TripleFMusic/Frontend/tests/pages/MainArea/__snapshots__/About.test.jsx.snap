// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="about-body"
  >
    <div
      class="icons-menu"
    >
      <div
        style="height: 3rem;"
      />
      <div
        class="icon"
      >
        <img
          alt="About"
          src="test-file-stub"
        />
        <span>
          About
        </span>
      </div>
      <div
        class="buffer"
      />
      <div
        class="icon"
      >
        <img
          alt="Home"
          src="test-file-stub"
        />
        <span>
          Home
        </span>
      </div>
      <div
        class="buffer"
      />
      <div
        class="icon"
      >
        <img
          alt="Music Gallery"
          src="test-file-stub"
        />
        <span>
          Music Gallery
        </span>
      </div>
      <div
        class="buffer"
      />
      <div
        class="icon"
      >
        <img
          alt="Upload new song"
          src="test-file-stub"
        />
        <span>
          Upload new song
        </span>
      </div>
      <div
        class="buffer"
      />
      <div
        class="icon"
      >
        <img
          alt="Internet Explorer"
          src="test-file-stub"
        />
        <span>
          Internet Explorer
        </span>
      </div>
    </div>
    <div
      class="space"
    />
    <div
      class="draggable-aboutWindow react-draggable"
      style="transform: translate(420px,0px);"
    >
      <div
        class="sc-lcIQwB brMtzx about-window"
      >
        <div
          class="sc-jxOSYQ cXImuS about-window-header"
        >
          <span>
            About TripleF Music
          </span>
          <button
            class="sc-dcJtft gGzOWQ"
            type="button"
            variant="default"
          >
            <span
              class="about-close-icon"
            />
          </button>
        </div>
        <div
          class="sc-fhzEvr heDfsD about-window-content"
        >
          <div
            class="sc-dhKdPU gySDZy"
            style="height: 550px;"
          >
            <div
              class="sc-kpDprT dqFqDm"
            >
              <div
                class="about-wrapper"
              >
                <div
                  class="sc-bbSYpP cmzDSq about-section"
                  variant="well"
                >
                  <h2
                    class="about-header"
                  >
                    About us!
                  </h2>
                  <div
                    class="sc-dhKdPU gySDZy about-text"
                  >
                    <div
                      class="sc-kpDprT dqFqDm"
                    >
                      <p>
                        Welcome to TripleF Music! We are a passionate team of three students from HFT Stuttgart, dedicated to bringing you a unique and engaging music experience. Our journey began with a shared love for music and technology, driving us to create a platform where users can explore, discover, and enjoy music in an oldschool and exciting way. Join us on our journey to revolutionize the way you experience music.
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="sc-bbSYpP cmzDSq about-section"
                  variant="well"
                >
                  <h2
                    class="about-header"
                  >
                    Our vision and goal
                  </h2>
                  <div
                    class="sc-dhKdPU gySDZy about-text"
                  >
                    <div
                      class="sc-kpDprT dqFqDm"
                    >
                      <p>
                        We aim to bring back the old-school nostalgia, giving users a delightful time-travel experience to the days when life was simpler and music had that special charm. Our main goal? Well, it's to pass our module, to be honest! But hey, a bit of creativity and fun never hurt anyone, right? We're here to combine our passion for retro vibes with modern technology, crafting a unique platform that makes discovering and enjoying music feel like a nostalgic adventure. Join us on this whimsical journey as we blend the best of the past with the present!
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="sc-bbSYpP cmzDSq about-section"
                  variant="well"
                >
                  <h2
                    class="about-header"
                  >
                    Our team
                  </h2>
                  <div
                    class="sc-dhKdPU gySDZy team-cutout"
                  >
                    <div
                      class="sc-kpDprT dqFqDm"
                    >
                      <div
                        class="team-member"
                      >
                        <img
                          alt="Fatima"
                          src="test-file-stub"
                        />
                        <p>
                          Fatima
                        </p>
                      </div>
                      <div
                        class="team-member"
                      >
                        <img
                          alt="Funda"
                          src="test-file-stub"
                        />
                        <p>
                          Funda
                        </p>
                      </div>
                      <div
                        class="team-member"
                      >
                        <img
                          alt="Fiona"
                          src="test-file-stub"
                        />
                        <p>
                          Fiona
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="sc-bbSYpP cmzDSq about-section"
                  variant="well"
                >
                  <h2
                    class="about-header"
                  >
                    Our motivation
                  </h2>
                  <div
                    class="sc-dhKdPU gySDZy"
                  >
                    <div
                      class="sc-kpDprT dqFqDm"
                    >
                      <div
                        class="tooltip-button"
                      >
                        <button
                          class="sc-dcJtft gGzOWQ"
                          type="button"
                          variant="default"
                        >
                          Matze
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="sc-bbSYpP cmzDSq about-section"
                  variant="well"
                >
                  <h2
                    class="about-header"
                  >
                    Contact
                  </h2>
                  <div
                    class="sc-dhKdPU gySDZy contact-cutout"
                  >
                    <div
                      class="sc-kpDprT dqFqDm"
                    >
                      <p>
                        Email: info@triplefmusic.de
                      </p>
                      <p>
                        Telefon: +49 171 56920382
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
